File: Algorithms/Searching/Binary Search/BinarySearchRecursion.js
function BinarySearchRecursion(arr,val,start=0,end=arr.length){
    let mid=Math.round((start+end)/2);
    if(start>end){
        return -1;
    }
    if(arr[mid]<val){
        return BinarySearchRecursion(arr,val,mid+1,end);
    }else if(arr[mid]>val){
        return BinarySearchRecursion(arr,val,start,mid-1);
    }else if(arr[mid] == val){
        return mid;
    } else {
        return -1;
    }
}


var arr = [1,2,4,5,7,8,12,63];
var value1 = 5,
value2 = 54,
value3 = 4;
console.log(`${value1} is at ${BinarySearchRecursion(arr,value1)}`);
console.log(`${value2} is at ${BinarySearchRecursion(arr,value2)}`);
console.log(`${value3} is at ${BinarySearchRecursion(arr,value3)}`);

File: Algorithms/Searching/Binary Search/BinarySearchRecursion.py

def BinarySearchRecursion(arr,val,start,end):
    mid = int((start+end)/2)
    if(start>end):
        return str(-1)
    
    if(arr[mid]>val):
        end = mid-1
        return BinarySearchRecursion(arr,val,start,end)
    elif(arr[mid]<val):
        start = mid+1
        return BinarySearchRecursion(arr,val,start,end)
    elif(arr[mid]==val):
        return str(mid)


arr = [1,5,4,2,7,8,2,12,63]
value1 = 7
value3 = 1
value2 = 45

print(str(value1) + ' is found at '+ BinarySearchRecursion(arr,value1,0,len(arr)))
print(str(value2) + ' is found at '+ BinarySearchRecursion(arr,value2,0,len(arr)))
print(str(value3) + ' is found at '+ BinarySearchRecursion(arr,value3,0,len(arr)))